pkg/kubelet/container/interface.go

```diff
@@ -0,0 +1,413 @@
+// operation succeeds.
+type PodSandboxManager interface {
+	// Create creates a sandbox based on the given config, and returns the
+	// the new sandbox.
+	Create(config *PodSandboxConfig) (PodSandboxID, error)
+	// Stop stops the sandbox by its ID. If there are any running
+	// containers in the sandbox, they will be terminated as a side-effect.
+	Stop(id PodSandboxID) error
+	// Delete deletes the sandbox by its ID. If there are any running
+	// containers in the sandbox, they will be deleted as a side-effect.
+	Delete(id PodSandboxID) error
+	// List lists existing sandboxes, filtered by the given PodSandboxFilter.
+	List(filter PodSandboxFilter) ([]PodSandboxListItem, error)
+	// Status gets the status of the sandbox by ID.
+	Status(id PodSandboxID) (PodSandboxStatus, error)
+}
+
+// PodSandboxConfig holds all the required and optional fields for creating a
+// sandbox.
+type PodSandboxConfig struct {
+	// Name is the name of the sandbox. The string should conform to
+	// [a-zA-Z0-9_-]+.
+	Name string
+	// Hostname is the hostname of the sandbox.
+	Hostname string
+	// DNSOptions sets the DNS options for the sandbox.
+	DNSOptions DNSOptions
+	// PortMappings lists the port mappings for the sandbox.
+	PortMappings []PortMapping
+	// Resources specifies the resource limits for the sandbox (i.e., the
+	// aggregate cpu/memory resources limits of all containers).
+	// Note: On a Linux host, kubelet will create a pod-level cgroup and pass
+	// it as the cgroup parent for the PodSandbox. For some runtimes, this is
+	// sufficent. For others, e.g., hypervisor-based runtimes, explicit
+	// resource limits for the sandbox are needed at creation time.
+	Resources PodSandboxResources
+	// Path to the directory on the host in which container log files are
+	// stored.
+	// By default the Log of a container going into the LogDirectory will be
+	// hooked up to STDOUT and STDERR. However, the LogDirectory may contain
+	// binary log files with structured logging data from the individual
+	// containers. For example the files might be newline seperated JSON
+	// structured logs, systemd-journald journal files, gRPC trace files, etc.
+	// E.g.,
+	//     PodSandboxConfig.LogDirectory = `/var/log/pods/<podUID>/`
+	//     ContainerConfig.LogPath = `containerName_Instance#.log`
+	//
+	// WARNING: Log managment and how kubelet should interface with the
+	// container logs are under active discussion in
+	// https://issues.k8s.io/24677. There *may* be future change of direction
+	// for logging as the discussion carries on.
+	LogDirectory string
+	// Labels are key value pairs that may be used to scope and select
+	// individual resources.
+	Labels Labels
+	// Annotations is an unstructured key value map that may be set by external
+	// tools to store and retrieve arbitrary metadata.
+	Annotations map[string]string
+
+	// Linux contains configurations specific to Linux hosts.
+	Linux *LinuxPodSandboxConfig
+}
```

Comment:
Since the `securityContext` of a container can have a `runAsUser`, we need something in this interface to be able to handle that. However, I think it should be solely part of a container's config, not pod sandbox.

Want to file a followup issue or make a PR to add this and discuss there instead?